parser.y

```diff
@@ -2132,15 +2132,25 @@ CreateViewStmt:
     {
 		startOffset := parser.startOffset(&yyS[yypt-1])
 		selStmt := $10.(*ast.SelectStmt)
-		selStmt.SetText(string(parser.src[startOffset:]))
+		selStmt.SetText(strings.TrimSpace(parser.src[startOffset:]))
 		x := &ast.CreateViewStmt {
  			OrReplace:     $2.(bool),
 			ViewName:      $7.(*ast.TableName),
 			Select:        selStmt,
+			Algorithm:     $3.(ast.ViewAlgorithm),
+			Definer:       $4.(*auth.UserIdentity),
+			Security:      $5.(ast.ViewSecurity),
 		}
 		if $8 != nil{
 			x.Cols = $8.([]model.CIStr)
 		}
+		if $11 !=nil {
+		    x.CheckOption = $11.(ast.ViewCheckOption)
+		    endOffset := parser.startOffset(&yyS[yypt])
+		    selStmt.SetText(strings.TrimSpace(parser.src[startOffset:endOffset]))
```

Comment:
I tested it locally. Seems we do not need this `strings.TrimSpace`.

Reply:
If I remove strings.TrimSpace, extra space would make test fail.
```
FAIL: parser_test.go:2220: testParserSuite.TestView

parser_test.go:2265:
    c.Assert(v.Select.Text(), Equals, "select c,d,e from t")
... obtained string = "" +
...     "select c,d,e from t \n" +
...     "                  "
... expected string = "select c,d,e from t"
```

Reply:
You can use sql string for test:
```go
src := `CREATE OR REPLACE ALGORITHM = UNDEFINED DEFINER = root@localhost
                  SQL SECURITY DEFINER
			      VIEW V(a,b,c) AS select c,d,e from t 
                  WITH CASCADED CHECK OPTION;`
```